clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -2958,47 +2983,171 @@ static bool checkWorkGroupSizeValues(Sema &S, Decl *D, const ParsedAttr &Attr,
   return Result;
 }
 
-// Handles reqd_work_group_size, work_group_size_hint and max_work_group_size
+template <typename WorkGroupAttrType>
+void Sema::addIntelSYCLTripleArgFunctionAttr(Decl *D,
```

Comment:
Given that you have to do the same checks for X, Y, and Z -- I wonder if it'd make sense for this to take an `ArrayRef<Expr *>` for the attribute arguments, and the caller would do: `{XDimExpr, YDimExpr, ZDimExpr}` to create the array ref.

Then this code can use a for loop to loop over the elements of the array and remove some duplication.

Reply:
I have created separate function to check for X, Y, and Z and removed duplicated codes.